File: 数组/多数元素.go
package 数组

//找到数组中的数量大于n/2的元素
func majorityElement(nums []int) int {

	seen := map[int]int{}
	for i := 0; i < len(nums); i++ {
		seen[nums[i]]++
		if seen[nums[i]] > len(nums)/2 {
			return nums[i]
		}
	}
	return -123
}

File: 链蒸蒸简单/多数元素.java
package 链蒸蒸简单;

public class 多数元素 {
    public int majorityElement(int[] nums) {
        int num = nums[0];
        int count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == num) {
                count++;
            } else {
                if (count == 0) {
                    num = nums[i];
                    count++;
                } else {
                    count--;
                }
            }
        }
        return num;
    }
}
